### PR TITLE
EasyFly v1.0

### PR DESCRIPTION
**Pull request by KaskMIL**
# In this new Pull Request, we as a team are adding the following features:
- Created a Front-End repository with the initial setup.
- Create the necessary reducers and slices to manage the creation of new Users, the creation and deletion of new Jets, and the creation and deletion of new Reservations.
- When the user runs the App, he can see the main page with a list of jets available, he can see the jet's details by clicking it, but if he wants to perform further actions must be logged-in.
- The user can create a new account on the Sign-in page.
- The user can create a new Reservation on the Reservation page, selecting a city, selecting a jet and a date and then confirming the reservation.
- The user can create a new Reservation from the Jet's details page by clicking on the reserve button. This button redirects to the Reservation Page, where the user must select the city, the jet it's fixed from the details page so the user only need to select the dates and then confirm the Reservation.
- The user can see a list of all his reservations on the My Reservations Page. The user can cancel a Reservation by clicking the Cancel Reservation button and confirming the cancellation.
- The user can delete a Jet on the Delete Jet Page. The user can see a list of all jets and delete a Jet by clicking the delete button and confirming the deleting.
- The user can finish his session by clicking the Logout button.

## Live Demo: [EasyFly](https://timely-sprinkles-616361.netlify.app/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,57 +1,78 @@
...
+import { getReservationsThunk } from './redux/reservation/reservationSlice';
+import {
+  authenticateThunk,
+  logout,
+  setUnAuthenticated,
+} from './redux/users/userSlice';
 import './App.css';
+import 'react-toastify/dist/ReactToastify.css';
 
 function App() {
```
Comment (vichuge) [inline src/App.js:23]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/arrows/LeftArrow.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './arrows.scss';
+
+function LeftArrow(props) {
```
Comment (vichuge) [inline src/components/arrows/LeftArrow.js:5]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/arrows/RightArrow.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './arrows.scss';
+
+function RightArrow(props) {
```
Comment (vichuge) [inline src/components/arrows/RightArrow.js:5]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/jet/DeleteJet.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { toast } from 'react-toastify';
+import { deleteJet } from '../../redux/jets/jetAPI';
+import { getJetsThunk } from '../../redux/jets/jetSlice';
+import './DeleteJet.scss';
+import AlertDialog from '../myReservation/Alert';
+
+function DeleteJet(props) {
```
Comment (vichuge) [inline src/components/jet/DeleteJet.js:10]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/jet/Jet.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import './Jet.scss';
+import * as FaIcons from 'react-icons/fa';
+import { IconContext } from 'react-icons';
+
+function Jet(props) {
```
Comment (vichuge) [inline src/components/jet/Jet.js:8]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/jet/JetDetails.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useParams, Link, useNavigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import './JetDetails.scss';
+import { IoChevronBackCircle } from 'react-icons/io5';
+import { BsFillCalendarCheckFill } from 'react-icons/bs';
+
+function JetDetails() {
```
Comment (vichuge) [inline src/components/jet/JetDetails.js:8]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/myReservation/Alert.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import DeleteIcon from '@mui/icons-material/Delete';
+
+import './alert.scss';
+
+export default function AlertDialog(props) {
```
Comment (vichuge) [inline src/components/myReservation/Alert.js:13]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/myReservation/MyReservationCard.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Box } from '@mui/material';
+import { toast } from 'react-toastify';
+import AlertDialog from './Alert';
+import { deleteReservation } from '../../redux/reservation/reservationAPI';
+import { getReservationsThunk } from '../../redux/reservation/reservationSlice';
+import './myReservationCard.scss';
+
+function MyReservationCard(props) {
```
Comment (vichuge) [inline src/components/myReservation/MyReservationCard.js:11]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/nav/Nav.js [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+import { Typography } from '@mui/material';
+import { useSelector, useDispatch } from 'react-redux';
+import { IconContext } from 'react-icons';
+
+import { authStatus, logout } from '../../redux/users/userSlice';
+import { cleanReservations } from '../../redux/reservation/reservationSlice';
+
+import './Nav.scss';
+
+function getWindowSize() {
```
Comment (vichuge) [inline src/components/nav/Nav.js:21]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/nav/Nav.js [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+  const { innerWidth } = window;
+  return { innerWidth };
+}
+
+function Nav({ children }) {
```
Comment (vichuge) [inline src/components/nav/Nav.js:26]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/nav/Nav.js [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+  const [sidebar, setSidebar] = useState(false);
+  const [windowSize, setWindowSize] = useState(getWindowSize());
+  const { name, status: userStatus } = useSelector((state) => state.user);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  useEffect(() => {
+    function handleWindowResize() {
```
Comment (vichuge) [inline src/components/nav/Nav.js:33]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/reservationForm/ConfirmData.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './confirmData.scss';
+import BasicTable from './Table';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+function ConfirmData(props) {
```
Comment (vichuge) [inline src/components/reservationForm/ConfirmData.js:7]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/reservationForm/ReservatinFormNew.js [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+import { toast } from 'react-toastify';
+import ConfirmData from './ConfirmData';
+import { postReservation } from '../../redux/reservation/reservationAPI';
+import { addReservation } from '../../redux/reservation/reservationSlice';
+
+import './ReservationFormNew.scss';
+import 'react-datepicker/dist/react-datepicker.css';
+
+function ReservationFormNew(props) {
```
Comment (vichuge) [inline src/components/reservationForm/ReservatinFormNew.js:14]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/reservationForm/Table.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+
+function createData(title, data) {
```
Comment (vichuge) [inline src/components/reservationForm/Table.js:10]: - Why don't use arrow func here? Is a better practice than function word.

File: src/components/reservationForm/Table.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+  return {
+    title, data,
+  };
+}
+
+export default function BasicTable(props) {
```
Comment (vichuge) [inline src/components/reservationForm/Table.js:16]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/deleteJet/DeleteJetPage.js [hunk 16]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import DeleteJet from '../../components/jet/DeleteJet';
+import Nav from '../../components/nav/Nav';
+import './DeleteJetPage.scss';
+
+function DeleteJetPage() {
```
Comment (vichuge) [inline src/pages/deleteJet/DeleteJetPage.js:7]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/main/Main.js [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+import Slider from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+import Jet from '../../components/jet/Jet';
+import './main.scss';
+import LeftArrow from '../../components/arrows/LeftArrow';
+import RightArrow from '../../components/arrows/RightArrow';
+import Nav from '../../components/nav/Nav';
+
+function Main() {
```
Comment (vichuge) [inline src/pages/main/Main.js:13]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/myReservation/MyReservationPage.js [hunk 18]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import Container from '@mui/material/Grid';
+import { Box, Typography } from '@mui/material';
+import MyReservationCard from '../../components/myReservation/MyReservationCard';
+import Nav from '../../components/nav/Nav';
+import './myReservation.scss';
+
+function MyReservation() {
```
Comment (vichuge) [inline src/pages/myReservation/MyReservationPage.js:9]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/reservation/ReservationPage.js [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+import { toast } from 'react-toastify';
+import ReservationFormNew from '../../components/reservationForm/ReservatinFormNew';
+import Nav from '../../components/nav/Nav';
+import cityData from './cityData';
+import handleUserStatus from '../../utils/UserStatusHandling';
+import { logout } from '../../redux/users/userSlice';
+
+import './reservationPage.scss';
+
+function ReservationPage() {
```
Comment (vichuge) [inline src/pages/reservation/ReservationPage.js:13]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/user/LoginPage.js [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+    y: 0,
+    opacity: 1,
+    transition: {
+      duration: 0.6,
+      ease: easing,
+    },
+  },
+};
+
+function LoginPage() {
```
Comment (vichuge) [inline src/pages/user/LoginPage.js:51]: - Why don't use arrow func here? Is a better practice than function word.

File: src/pages/user/SignupPage.js [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+    y: 0,
+    opacity: 1,
+    transition: {
+      duration: 0.6,
+      ease: easing,
+    },
+  },
+};
+
+function SignupPage() {
```
Comment (vichuge) [inline src/pages/user/SignupPage.js:52]: - Why don't use arrow func here? Is a better practice than function word.

**Review by vichuge — COMMENTED** [review]
Hi dear student, 

Good job so far!
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!

mario_ladders


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Meltrust — APPROVED** [review]
#### Hi @KaskMIL,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- All functions have been refactored to ES6✔️ 


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Pull request merged**